src/Data/Vec/Binary/Base.agda

```diff
@@ -0,0 +1,104 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Vectors with faster indexing based on binary numbers
+------------------------------------------------------------------------
+
+{-# OPTIONS --cubical-compatible --safe #-}
+
+module Data.Vec.Binary.Base where
+
+open import Data.Fin.Binary.Base as Finᵇ hiding (zero; suc)
+open import Data.Nat.Binary.Base
+open import Function.Base using (_∘_)
+open import Level using (Level)
+
+private
+  variable
+    a b c : Level
+    A : Set a
+    B : Set b
+    C : Set c
+    n : ℕᵇ
+
+-- Core type
+------------------------------------------------------------------------
+
+data Vecᵇ (A : Set a) : ℕᵇ → Set a where
```

Comment:
To me this looks more like some sort of balanced binary tree in disguise rather than a vector encoding? I'm a bit reluctant to add it under the `Vec` hierarchy...

Reply:
It's a vector implemented as a binary tree. It's based, roughly, on zeroless binary random access lists from section 9.2.2 of Okasaki's purely functional data structures. It has operations for manipulating the head of the vector, and none for inserting or deleting from arbitrary positions (those are hard to write). It feels a lot more vector-like than tree-like to me, in terms of interface. Is there anything I could define that would convince you it's truly a vector?

Reply:
Sorry that was poorly phrased on my part. It's not that I don't believe there's an interface that is vector-like (although the way you describe it with only being able to manipulate the head, it sounds a little bit more like a list). It's that maybe there's a (slightly more general?) tree data-structure that is lurking underneath here that we should be able to extract out?

Reply:
I really don't think there is a useful tree structure we can extract here! The shape of the internal tree is guided carefully by the `ℕᵇ` index so that there's only one possible shape for each size.

I think you may have slightly misunderstood me when I said that we can't insert or delete from arbitrary positions. We can in fact efficiently perform arbitrary lookups (that the whole point of this type, really), and we can do `updateAt` quickly.